circuit_knitting/cutting/cutting_evaluation.py

```diff
@@ -238,12 +239,80 @@ def _append_measurement_circuit(
     return qc
 
 
+def generate_cutting_experiments(
+    circuits: QuantumCircuit | dict[str | int, QuantumCircuit],
+    observables: PauliList | dict[str | int, PauliList],
+    num_samples: int | float,
+) -> tuple[
+    list[QuantumCircuit] | dict[str | int, list[QuantumCircuit]],
+    list[tuple[float, WeightType]],
+]:
+    """
+    Generate cutting subexperiments and their associated weights.
+
+    If the input, ``circuits``, is a :class:`QuantumCircuit` instance, the
+    output subexperiments will be contained within a 1D array, and ``observables`` is
+    expected to be a :class:`PauliList` instance.
+
+    If the input circuit and observables are specified by dictionaries with partition labels
+    as keys, the output subexperiments will be returned as a dictionary which maps a
+    partition label to to a 1D array containing the subexperiments associated with that partition.
+
+    In both cases, the subexperiment lists are ordered as follows:
+        :math:`[sample_{0}observable_{0}, sample_{0}observable_{1}, ..., sample_{0}observable_{N}, ..., sample_{M}observable_{N}]`
```

Comment:
I find this a bit more clear:

```suggestion
        :math:`[sample_{0}observable_{0}, ..., sample_{0}observable_{N}, sample_{1}observable_{0}, ..., sample_{M}observable_{N}]`
```

Also, not sure if the ellipses render correctly or funny in Sphinx (I haven't checked), but if I were writing a paper I'd use `\ldots` here instead of `...`.